conan/internal/cache/db/cache_database.py

```diff
@@ -61,11 +59,9 @@ def remove_build_id(self, pref):
         self._packages.remove_build_id(pref)
 
     def get_matching_build_id(self, ref, build_id):
-        # TODO: This can also be done in a single query in DB
-        for d in self._packages.get_package_references(ref):
-            existing_build_id = d["build_id"]
-            if existing_build_id == build_id:
-                return d["pref"]
+        result = self._packages.get_package_references_with_build_id_match(ref, build_id)
```

Comment:
Looks good, have you estimated the time savings with this one?

Reply:
Yes, for 20 runs of `pytest test/integration/package_id/build_id_test.py --count=20` , I see these `time` results:

Before:

```

________________________________________________________
Executed in   16.80 secs    fish           external
   usr time   11.01 secs    0.08 millis   11.01 secs
   sys time    3.88 secs    1.48 millis    3.88 secs

```

After:

```

________________________________________________________
Executed in   16.51 secs    fish           external
   usr time   10.88 secs    0.08 millis   10.88 secs
   sys time    3.70 secs    1.60 millis    3.70 secs

```

Reply:
Seems really small improvement, I'd expect a bit more?

Reply:
Before:

```
$ time pytest test/integration/package_id/build_id_test.py::TestBuildIdTest::test_create --count=100
________________________________________________________
Executed in   12.86 secs    fish           external
   usr time    8.13 secs    0.08 millis    8.13 secs
   sys time    3.14 secs    1.74 millis    3.14 secs
```


After:

```
$ time pytest test/integration/package_id/build_id_test.py::TestBuildIdTest::test_create --count=100

Executed in   12.52 secs    fish           external
   usr time    7.93 secs    0.10 millis    7.93 secs
   sys time    2.92 secs    2.50 millis    2.92 secs
```